crates/tests/fixtures/files/arrays.fe

```diff
@@ -1,10 +1,47 @@
 use std::evm
 
+const ZERO_ADDRESS: address = 0x0000000000000000000000000000000000000000
+
+struct AddressArray {
+    array: Array<address, 2>
+    length: u256
+
+    pub fn new() -> AddressArray {
+        return AddressArray (
+            array: [ZERO_ADDRESS; 2],
+            length: 0
+        )
+    }
+
+    pub fn push(mut self, value: address) {
+        // Omitting bounds check for simplicity
+        self.array[self.length] = value
+        self.length += 1
+    }
+
+    pub fn length(self) -> u256 {
+        return self.length
+    }
+
+    pub fn get(self, index: u256) -> address {
+        return self.array[index]
+    }
+}
+
 contract Foo {
     my_array: Array<u256, 10>
+    addr_array: AddressArray
 
     pub fn __init__(mut self) {
         self.my_array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+        let mut addr_array: AddressArray = AddressArray::new()
+        addr_array.push(value: 0x627306090abaB3A6e1400e9345bC60c78a8BEf57)
+        addr_array.push(value: 0xf17f52151EbEF6C7334FAD080c5704D77216b732)
+        self.addr_array = addr_array
+    }
+
+    pub fn get_address_from_storage(self, index: u256) -> address {
+        return self.addr_array.get(index)
```

Comment:
The test works if we change that to be `self.addr_array.to_mem().get(index)`. I believe the compiler rejected that in the past but doesn't anymore.